### PR TITLE
aragonAPI docs restructure

### LINKED ISSUE #102: Restructure aragonAPI section so that the relationships are more clear
See https://forum.aragon.org/t/on-aragon-naming/203/12:

> Currently, when clicking on aragonAPI in the developer portal, one is redirected to the aragonJS root page. The first package mentioned is @aragon/client, all the while the repository is aragon.js.

We should create a different landing page for aragonAPI itself, that points to aragon.js and `@aragon/app` as being the JavaScript implementations of aragonAPI.

### PR DESCRIPTION
**Pull request by 0xGabi**
closes #102 

TODO:

- [x] Merge Docusaurus PR (https://github.com/aragon/Docusaurus/pull/4) and update dependency

- [x] Merge fixes in aragon.js (https://github.com/aragon/aragon.js/pull/271)

- [x] Update `git-ref` of aragon.js sync script to master

- [x] Add styles for subcategories

### CODE REVIEW

File: docs/api-intro.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+hide_title: true
+---
+
+![](/docs/assets/brand/aragonapi.png)
+
+Standard set of APIs and specifications used to interact with aragonOS-powered contracts by handling transaction pathing, upgradeability, and contract state without depending of a centralized service.
+
+## Reference implementations in specific languages
+
+### aragonJS
```
Comment (sohkai) [inline docs/api-intro.md]:
```suggestion
### JavaScript
```

"aragonJS" isn't really a name we use. Could we instead change this section to be "JavaScript" so we can hide "aragon.js"?

File: docs/api-intro.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+A JavaScript implementation of aragonAPI, used to interact with aragonOS by handling transaction pathing, upgradeability, identity providers and state of the contracts.
+
+Some of the things you can do with aragonJS are:
```
Comment (sohkai) [inline docs/api-intro.md]:
```suggestion
Some of the things you can do with the JavaScript implementation are:
```

File: docs/api-intro.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+![](/docs/assets/brand/aragonapi.png)
+
+Standard set of APIs and specifications used to interact with aragonOS-powered contracts by handling transaction pathing, upgradeability, and contract state without depending of a centralized service.
+
+## Reference implementations in specific languages
+
+### JavaScript
+
+A JavaScript implementation of aragonAPI, used to interact with aragonOS by handling transaction pathing, upgradeability, identity providers and state of the contracts.
```
Comment (2color) [inline docs/api-intro.md]:
```suggestion
A JavaScript implementation of the aragonAPI, used to interact with aragonOS by handling transaction pathing, upgradeability, identity providers and state of the contracts.
```

File: docs/api-intro.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Some of the things you can do with the JavaScript implementation are:
+
+- Connect contracts to front-end
+- Interact with the EVM and [aragonOS](os-intro.md) when using the full Aragon System
+- Abstract Web3 so a direct Web3 connection isn't exposed to the browser
```
Comment (2color) [inline docs/api-intro.md]:
```suggestion
- Interact with contracts directly – an abstraction over web3.js
```

File: docs/api-intro.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+- Use a store method wich mantains the application state using caching
```
Comment (2color) [inline docs/api-intro.md]: Provide access to application state with built in client side caching

Comment (2color) [inline docs/api-intro.md]:
```suggestion
- Get access to application state with built in client-side caching
```

**Review by 2color — APPROVED** [review]
Nice! Besides a couple of small suggestions, this is moving in the right direction.

File: docs/api-intro.md [hunk 6]
```diff
@@ -29,9 +29,9 @@ Some of the things you can do with the JavaScript implementation are:
 
 #### Docs
 
+- [React API](js-ref-react.md)
```
Comment (sohkai) [inline docs/api-intro.md]: I'd probably move this as a subcategory of the App API

File: docs/api-intro.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+### JavaScript
+
+A JavaScript implementation of the aragonAPI, used to interact with aragonOS by handling transaction pathing, upgradeability, identity providers and state of the contracts.
+
+Some of the things you can do with the JavaScript implementation are:
+
+- Connect contracts to front-end
+- Interact with the EVM and [aragonOS](os-intro.md) when using the full Aragon System
+- Interact with contracts directly – an abstraction over web3.js
```
Comment (sohkai) [inline docs/api-intro.md]: I'd combine these two lines into one:

```
- Interact with aragonOS and aragonOS-powered applications directly through an abstraction over web3.js
```

File: docs/api-intro.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+- Get access to application state with built in client-side caching
+- Pull [Radspec](human-readable-txs.md) notices from contracts
```
Comment (sohkai) [inline docs/api-intro.md]:
```suggestion
- Create human-readable transaction descriptions for your smart contracts through [Radspec](human-readable-txs.md)
```

**Pull request merged**